### PR TITLE
Add fuel cost time series parameters methods

### PR DESCRIPTION
**Pull request by rodrigomha**

### CODE REVIEW

File: src/parameters/add_parameters.jl [hunk 1]
```diff
@@ -256,7 +273,51 @@ function _add_parameters!(
     U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}},
     W <: AbstractDeviceFormulation,
 } where {D <: PSY.Component}
-    _add_time_series_parameters!(container, param, devices, model)
+    ts_type = get_default_time_series_type(container)
+    if !(ts_type <: Union{PSY.AbstractDeterministic, PSY.StaticTimeSeries})
+        error("add_parameters! for TimeSeriesParameter is not compatible with $ts_type")
```
Comment (jd-lara) [inline src/parameters/add_parameters.jl:278]:
```suggestion
        error("add_parameters! for ObjectiveFunctionParameter is not compatible with $ts_type")
```

File: src/devices_models/devices/common/add_to_expression.jl [hunk 2]
```diff
@@ -1533,6 +1571,208 @@ function add_to_expression!(
...
+                    fuel_expr,
+                )
+            end
+        elseif value_curve isa PSY.QuadraticCurve
+            error("Quadratic Curves are not accepted with Compact Formulation: $W")
+        end
+    end
+end
+
+#TODO: FuelConsumption for PowerAboveMinimumVariable
```
Comment (jd-lara) [inline src/devices_models/devices/common/add_to_expression.jl:1710]: @rodrigomha can we delete this code?

File: src/parameters/update_cost_parameters.jl [hunk 3]
```diff
@@ -14,31 +14,62 @@ function _update_parameter_values!(
...
                     component,
-                    t,
+                    PSY.get_operation_cost(component);
+                    start_time = initial_forecast_time,
+                    len = horizon,
+                )
+                variable_cost_forecast_values = TimeSeries.values(ts_vector)
+                for (t, value) in enumerate(variable_cost_forecast_values)
+                    if attributes.uses_compact_power
+                        # TODO implement this
```
Comment (jd-lara) [inline src/parameters/update_cost_parameters.jl:31]: @rodrigomha is this TODO done?

File: src/parameters/update_cost_parameters.jl [hunk 4]
```diff
@@ -14,31 +14,62 @@ function _update_parameter_values!(
...
+                fuel_curve = PSY.get_variable(op_cost)
+                ts_vector = PSY.get_fuel_cost(
+                    component;
+                    start_time = initial_forecast_time,
+                    len = horizon,
+                )
+                fuel_cost_forecast_values = TimeSeries.values(ts_vector)
+                for (t, value) in enumerate(fuel_cost_forecast_values)
+                    if attributes.uses_compact_power
+                        # TODO implement this
```
Comment (jd-lara) [inline src/parameters/update_cost_parameters.jl:55]: same here

**Review by jd-lara — COMMENTED** [review]
@rodrigomha the code in here is now compatible with multiple `has_time_series` checks and also resolved the mixed expression type. It is missing QA/QC and cleaning up the TODOs

File: src/parameters/update_cost_parameters.jl (hunk 4 above)
Comment (rodrigomha) [reply src/parameters/update_cost_parameters.jl:55]: These TODOs were there before this PR. Not sure if we can safely remove those TODO.

**Pull request closed without merging**